woocommerce/woocommerce-gutenberg-products-block
# Add an endpoint for processing pay for order orders

Add an endpoint for processing pay-for-order orders so we can pay for the orders.

Fixes #9312 

#### Other Checks



### Testing instructions
* Test checkout as a guest
<img width="892" alt="Screen Shot 2023-07-20 at 8 33 51 AM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/ff17ec89-b425-45c3-956b-cd4d206aaa0a">


1. Install and activate a plugin with pay-for-order flow. ex WooCommerce-Bookings
2. Create a product that needs confirmation and purchase the product
3. Go to WooCommerce -> Orders
4. Make sure the Customer is `Guest`
5. Click the `Customer payment page` to go to the pay-for-order page
6. Copy the `key` from the URL
7. Use Postman to test this endpoint `http://merchant.local/wp-json/wc/store/checkout/YOUR_ORDER_NUMBER?key=YOUR_KEY&billing_email=YOUR_BILLING_EMAIL`
8. Make sure on Postman the authorization type set to `No Auth`
9. POST to checkout the order. ex:
```
{
  "billing_address": {
        "first_name": "Test",
        "last_name": "TestBilling",
        "company": "",
        "address_1": "123 test",
        "address_2": "",
        "city": "San Diego",
        "state": "CA",
        "postcode": "92103",
        "country": "US",
        "email": "test@hacker.com",
        "phone": ""
    },
    "shipping_address": {
        "first_name": "Test",
        "last_name": "Test Last",
        "company": "",
        "address_1": "123 test",
        "address_2": "",
        "city": "San Diego",
        "state": "CA",
        "postcode": "92103",
        "country": "US",
        "email": "test@hacker.com",
        "phone": ""
    },
  "payment_method": "cod"
}
```

* Test checkout as an existing customer
<img width="908" alt="Screen Shot 2023-07-20 at 8 34 30 AM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/8802832e-0768-42f5-8da0-a184de3f7cb7">

1. Go back to WooCommerce->Orders and use the same order you created above
2. Set status: `Pending payment` and Customer: `Existing customer`
3. Go to WooCommerce -> Settings -> Advanced -> Rest API
4. Add key with the customer email, read/write permission, copy the Consumer key and secret to Postman
5. On Postman, change the authorization type to OAuth 1.0, and paste the key and secret
6. Remove the `key` and `billing_email`params and send the request
7. Send the request again and should see below message
<img width="623" alt="Screen Shot 2023-07-20 at 11 22 39 AM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/7ecaa348-c2f4-4762-bb95-8bae8a463859">


#### Automated Tests




### Changelog

> Add an endpoint for processing pay-for-order orders.
